csi.proto

```diff
@@ -172,6 +206,50 @@ message CreateVolumeRequest {
   // logged, etc.) by the CO.
   // This field is OPTIONAL.
   map<string, string> controller_create_secrets = 5;
+
+  // Contains all attributes of the source that is used to create
+  // the volume from. This field is OPTIONAL.
+  ContentSource content_source = 6;
+}
+
+// Specifies what source the volume will be created from. One of the
+// source_type fields MUST be specified. Only SnapshotReference is
+// a supported ContentSource at the moment.
+message ContentSource {
+  oneof source_type {
+    SnapshotReference snapshot_ref = 1;
+  }
+}
+
+// The information about a snapshot reference as a source to create
+// a volume from.
+message SnapshotReference {
```

Comment:
Ok, I'll create a nested type to avoid confusion.

So CO wants to create a volumeB from this SnapshotReference.  CO needs to know how much space is needed to create volumeB from this snapshot so "size" is needed.  Regarding "volume_id", that is the original volume where this snapshot is created from.